lib/seadsa/ShadowMem.cc

```diff
@@ -1367,6 +1379,46 @@ void ShadowMemImpl::visitSetShadowMem(CallBase &CB) {
   associateConcretePtr(memDef, ptr, &callInst);
 }
 
+void ShadowMemImpl::visitBorMem2Reg(CallBase &CB) {
+  LOG("dsa.bormem2reg", errs() << "Visiting bor_mem2reg \n");
+  /* isMetadataSet definition:
+     bool @sea_is_alloc(i8*)
+     operand 0 is the pointer to be marked
+  */
+  auto &src_ptr = *CB.getArgOperand(0);
+  if (!m_graph->hasCell(src_ptr)) {
+    LOG("dsa.bormem2reg", errs() << "no cell for: " << src_ptr << "\n");
+    return;
+  }
+  const dsa::Cell &cell = m_graph->getCell(src_ptr);
+  if (cell.isNull()) {
+    LOG("dsa.bormem2reg", errs() << "cell is null for: " << src_ptr << "\n");
+    return;
+  }
+  auto &callInst = CB;
+  m_B->SetInsertPoint(&callInst);
+  CallInst &memUse = mkShadowLoad(*m_B, cell, 1 /* bytes to access */);
+  associateConcretePtr(memUse, src_ptr, &callInst);
+}
+
+void ShadowMemImpl::visitMoveReg2Mem(CallBase &CB) {
+  LOG("dsa.setshadowmem", errs() << "Visiting MoveReg2Mem \n");
+  auto &callInst = CB;
+  auto &ptr = *CB.getArgOperand(1);
+  if (!m_graph->hasCell(ptr)) {
+    LOG("dsa.movereg2mem", errs() << "no cell for: " << ptr << "\n");
+    return;
+  }
+  const dsa::Cell &cell = m_graph->getCell(ptr);
+  if (cell.isNull()) {
+    LOG("dsa.movereg2mem", errs() << "cell is null for: " << ptr << "\n");
+    return;
+  }
+  m_B->SetInsertPoint(&callInst);
+  CallInst &memDef = mkShadowStore(*m_B, cell, 1 /* bytes to access */);
```

Comment:
Should MoveReg2Mem simply create pointer aliasing?

Reply:
fixed.

Reply:
fyi @LinerSu